app/components/combined_system_message/combined_system_message.js

```diff
@@ -327,16 +336,23 @@ export default class CombinedSystemMessage extends React.PureComponent {
                     userIds = [this.props.currentUserId];
                 } else {
                     // Not something the current user did or was affected by
-                    continue;
+                    return;
                 }
             }
 
-            content.push(
-                <React.Fragment key={postType + actorId}>
-                    {this.renderFormattedMessage(postType, userIds, actorId, {baseText: style.baseText, linkText: style.linkText})}
-                </React.Fragment>
-            );
-        }
+            if (postType === REMOVE_FROM_CHANNEL) {
+                removedUserIds = removedUserIds.concat(userIds);
+            } else if (postType !== REMOVE_FROM_CHANNEL && removedUserIds.length > 0) {
+                content.push(this.renderMessage(postType, removedUserIds, actorId, style));
```

Comment:
This is trying to render a non-REMOVE_FROM_CHANNEL message with the IDs of the users that were removed in the previous messages

Reply:
I'm trying to think of a better way to do this since it seems like the problem here is that this loop is really complicated because we now keep REMOVE_FROM_CHANNEL messages from different users separate like we do with ADD_TO_CHANNEL posts, but we render REMOVE_FROM_CHANNEL posts combined in the UI.

I think the ideal fix would be to change the REMOVE_FROM_CHANNEL posts to be like "User removed from channel by OtherUser" since then we wouldn't need to recombine them, but that's a lot of work. What if instead we just had REMOVE_FROM_CHANNEL messages always show up last? Then we could remove most of this if statement and just have this loop be like
```
content = []
removedIds = []
for (message in messageData) {
    if (message.type == REMOVE_FROM_CHANNEL) {
        removedIds.push(message.userIds)
    } else {
        content.push(renderMessage(message));
    }
}

if removedIds.length > 0 {
    content.push(renderMessage({type: REMOVE_FROM_CHANNEL, userIds: removedIds}));
}
```

Reply:
1.  Thanks for pointing out.  You're right.  I'll just wait on item 2 on how we'll proceed.
2.  @esethna Could you please confirm whether we can put the order of REMOVE_FROM_CHANNEL messages to show up last?  Currently, it's set second to the last.  See sort order below:
```
    const postTypePriority = {
        [Posts.POST_TYPES.JOIN_TEAM]: 0,
        [Posts.POST_TYPES.ADD_TO_TEAM]: 1,
        [Posts.POST_TYPES.REMOVE_FROM_TEAM]: 2,
        [Posts.POST_TYPES.LEAVE_TEAM]: 3,
        [Posts.POST_TYPES.JOIN_CHANNEL]: 4,
        [Posts.POST_TYPES.ADD_TO_CHANNEL]: 5,
        [Posts.POST_TYPES.REMOVE_FROM_CHANNEL]: 6,
        [Posts.POST_TYPES.LEAVE_CHANNEL]: 7,
    };
```

Reply:
Ya that's fine with me.

Reply:
Thanks @hmhealey @esethna,  I've updated the logic.  Also I submitted a PR (https://github.com/mattermost/mattermost-redux/pull/586) to mattermost-redux but it's not blocking to how this change works.